File: kestrel-java/src/main/java/io/github/matthewjones372/kestrel/java/Rates.java
package io.github.matthewjones372.kestrel.java;

import io.github.matthewjones372.kestrel.Rate;

/**
 * How often virtual users arrive. Kotlin writes {@code 50.perSecond}, which is
 * an extension on a value class and so has a hash in the name it compiles to.
 */
public final class Rates {

    private Rates() {
    }

    public static Rate perSecond(double rate) {
        return (Rate) Boxes.perSecond(rate);
    }

    public static Rate perMinute(double rate) {
        return (Rate) Boxes.perMinute(rate);
    }
}

File: proofload-scala/src/main/scala/io/github/matthewjones372/proofload/scala/Rates.scala
package io.github.matthewjones372.proofload.scala

import io.github.matthewjones372.proofload.Rate
import io.github.matthewjones372.proofload.java.Rates

/**
 * How often virtual users arrive. Kotlin writes `50.perSecond`, which is an
 * extension on a value class and so compiles to a name with a hash in it.
 */
extension (rate: Int)

  def perSecond: Rate = Rates.perSecond(rate.toDouble)

  def perMinute: Rate = Rates.perMinute(rate.toDouble)

extension (rate: Double)

  def perSecond: Rate = Rates.perSecond(rate)

  def perMinute: Rate = Rates.perMinute(rate)
